api/v1beta1/azuremachine_webhook.go

```diff
@@ -152,6 +154,24 @@ func (m *AzureMachine) ValidateUpdate(oldRaw runtime.Object) error {
 		}
 	}
 
+	if !reflect.DeepEqual(m.Spec.NetworkInterfaces, old.Spec.NetworkInterfaces) {
+		// The defaulting webhook may have migrated values from the old SubnetName field to the new NetworkInterfaces format.
+		old.Spec.SetNetworkInterfacesDefaults()
+
+		// The reconciler will populate the SubnetName on the first interface if the user left it blank.
```

Comment:
This is running the Network Interfaces defaulter on the "old" resource. The defaulting webhook would have run against the "new" resource. By running the Network Interfaces defaulter against the "old" resource, we make the "old" and "new" identical in the case that the modification was due to the defaulter migrating values from the old SubnetName/AcceleratedNetworking fields to the new NetworkInterfaces fields. I used this as a way to bypass immutability for this very specific case since we need the validating webhook to permit this particular mutation.